src/System Application/App/File System/src/FileSystem/FileSystem.Codeunit.al

```diff
@@ -0,0 +1,252 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.FileSystem;
+
+codeunit 9454 "File System"
+{
+    InherentPermissions = X;
+    InherentEntitlements = X;
+
+    var
+        FileSystemImpl: Codeunit "File System Impl.";
+
+    /// <summary>
+    /// Initialized the File System for the given scenario.
+    /// </summary>
+    /// <param name="Scenario">File Scenario to use.</param>
+    procedure Initialize(Scenario: Enum "File Scenario")
+    begin
+        FileSystemImpl.Initialize(Scenario);
+    end;
+
+    /// <summary>
+    /// Initialized the File System for the give file account.
+    /// </summary>
+    /// <param name="FileAccount"> File Account to use.</param>
+    procedure Initialize(FileAccount: Record "File Account")
+    begin
+        FileSystemImpl.Initialize(FileAccount);
+    end;
+
+    /// <summary>
+    /// List all files from the given path.
+    /// </summary>
+    /// <param name="Path">Folder to list</param>
+    /// <param name="FilePaginationData">Defines the pagination data.</param>
+    /// <param name="FileAccountContent">File account content.</param>
+    procedure ListFiles(Path: Text; FilePaginationData: Codeunit "File Pagination Data"; var FileAccountContent: Record "File Account Content" temporary)
+    begin
+        FileSystemImpl.ListFiles(Path, FilePaginationData, FileAccountContent);
+    end;
+
+    /// <summary>
+    /// Retrieves a file from the file account.
+    /// </summary>
+    /// <param name="Path">File Path to open.</param>
+    /// <param name="Stream">Stream which contains the file content.</param>
+    [TryFunction]
```

Comment:
I'd prefer if we don't use `[TryFunction]` on facade layers but be more explicit that it returns a boolean.

Use the `[TryFunction]` in the implementation layer.

Reply:
Having a try function in the facade layer enables us to catch the error outside this module.
if the Error is catched internally then it's getting more complicated to handle errors outside.

Reply:
Makes sense. Keep it as is.